haystack/pipelines/base.py

```diff
@@ -619,6 +697,110 @@ def save_to_yaml(self, path: Path, return_defaults: bool = False):
+            f"_______________________________________________________"
+        )
+        return s
+
+    def _format_wrongs_node(self, node_name: str, wrongs_formatted: str):
+        s = (
+            f"                Wrong {node_name} Examples\n"
+            f"=======================================================\n"
+            f"{wrongs_formatted}\n"
+            f"=======================================================\n"
+        )
+        return s
+
+    def _format_wrongs_report(self, eval_result: EvaluationResult, n_wrong_examples: int = 3):
+        examples = {
+            node: eval_result.wrong_examples(node, doc_relevance_col="gold_id_or_answer_match", n=n_wrong_examples) 
+                for node in eval_result.node_results.keys()
+        }
+        examples_formatted = {
+            node: "\n".join([self._format_wrong(example) for example in examples]) 
+                for node, examples in examples.items()
+        }
+
+        return "\n".join([self._format_wrongs_node(node, examples) for node, examples in examples_formatted.items()])
+
+    def _format_pipeline_node(self, node: str, metrics: dict, metrics_top_1):
+        metrics = metrics.get(node, {})
+        metrics_top_1 = {f"{metric}_top_1": value for metric, value in metrics_top_1.get(node, {}).items()}
+        node_metrics = {**metrics, **metrics_top_1}
+        node_metrics_formatted = "\n".join(sorted([f"                        | {metric}: {value:5.3}" for metric, value in node_metrics.items()])) 
+        node_metrics_formatted = f"{node_metrics_formatted}\n" if len(node_metrics_formatted) > 0 else ""
+        s = (
+            f"                      {node}\n"
+            f"                        |\n"
+            f"{node_metrics_formatted}"
+            f"                        |"
+        )
+        return s
+
+    def _format_pipeline_overview(self, metrics: dict, metrics_top_1: dict):
+        pipeline_overview = "\n".join([self._format_pipeline_node(node, metrics, metrics_top_1) for node in self.graph.nodes])
+        s = (
+            f"================== Evaluation Report ==================\n"
+            f"=======================================================\n"
+            f"                   Pipeline Overview\n"
+            f"=======================================================\n"
+            f"{pipeline_overview}\n"
+            f"                      Output\n"
+            f"=======================================================\n"
+        )
+        return s
+
+    def print_eval_report(
+        self, 
+        eval_result: EvaluationResult, 
+        n_wrong_examples: int = 3, 
+        metrics_filter: Optional[Dict[str, List[str]]] = None):
+        """
+        Prints evaluation report containing a metrics funnel and worst queries for further analysis.
+        
+        :param eval_result: The evaluation result, can be obtained by running eval().
+        :param n_wrong_examples: The number of worst queries to show.
```

Comment:
Maybe we make that more generic to also include correct examples in a separate issue and pull request.